### PR TITLE
Build C++ SDK for Linux ARM64 

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Part of #4136

Had to use a buildjet runner because some of our `rerun_c` dependencies (`ring` foremost) had trouble cross compiling after all.

Artifact I built from CI by emulating push-to-main ci:
https://build.rerun.io/commit/04b1eb6/rerun_cpp_sdk.zip

Not properly tested, but evidently has arm64 linux static library in it.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5489/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5489/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5489/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5489)
- [Docs preview](https://rerun.io/preview/20c26b4682d0d874dddbaab26102b8a061a885a3/docs) 
- [Examples preview](https://rerun.io/preview/20c26b4682d0d874dddbaab26102b8a061a885a3/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by Wumpf** [comment]
`ring` crate has trouble cross compiling

**Review by abey79 — APPROVED** [review]
The cmake stuff if beyond me, but the rest appears really straightforward 👍🏻

File: rerun_cpp/CMakeLists.txt [hunk 1]
```diff
@@ -57,8 +57,13 @@ if(NOT TARGET rerun_c)
             message(WARNING "Unknown architecture ${CMAKE_SYSTEM_PROCESSOR}, can't find rerun_c library.")
         endif()
     elseif(UNIX) # if(LINUX) # CMake 3.25
-        # TODO(andreas): Arm support.
-        set(RERUN_C_LIB_DEFAULT ${RERUN_C_DEFAULT_LIB_DIR}/librerun_c__linux_x64.a)
+        if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
+            set(RERUN_C_LIB_DEFAULT ${RERUN_C_DEFAULT_LIB_DIR}/librerun_c__linux_x64.a)
+        elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "arm64")
```
Comment (abey79) [inline rerun_cpp/CMakeLists.txt:62]: I get an unknown "aarch64" architecture error. Half of the world call that arm64, the other aarch64. We must check all inputs against both versions imo.

**Review by abey79 — APPROVED** [review]
I take it back :) We must make sure to always accept `aarch64` and `arm64` in input interchangeably.

**Comment by Wumpf** [comment]
`python ./scripts/ci/bundle_and_upload_rerun_cpp.py --git-hash d548beb11a3579827e233ed9abee7b832e285589 --platform-filter linux-arm64 --skip-test` 

was used to create & upload a new linux arm only bundle. Tested by @abey79 on a debian VM against the opencv example. It works! \o/

**Pull request merged**